subsys/bluetooth/host/conn.c

```diff
@@ -1765,7 +1762,8 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 		 */
 		if (IS_ENABLED(CONFIG_BT_CENTRAL) &&
 		    conn->type == BT_CONN_TYPE_LE) {
-			k_delayed_work_submit(&conn->update_work, CONN_TIMEOUT);
+			k_delayed_work_submit(&conn->update_work,
+				bt_dev.create_param.timeout);
```

Comment:
Maybe we should validate this parameter? Setting it to zero would leave the initiator running.
```
	range 1 BT_RPA_TIMEOUT if BT_PRIVACY
	range 1 65535
```
This is the Kconfig range.

Reply:
Are there any other checks worth putting in a bt_conn_create_params_valid function?